src/Sylius/Bundle/ResourceBundle/Storage/CookieStorage.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Storage;
+
+use Sylius\Component\Resource\Storage\StorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class CookieStorage implements StorageInterface, EventSubscriberInterface
```

Comment:
But there are two reasons why could we change this class, so for me, it violates the SRP. 
Another thing, is that cookie storage is totally valid as a separate class. We use it in currency context. And this part is just a simple storage, which could be easly replaced. But the `though` logic should be excluded from this storage and could be placed in the cookie listener. And the cookie listener is only applicable to cookie storage, but implementation of cookie storage could be replaced.